### PR TITLE
Issue 08 07 abstract and branching process model

### PR DESCRIPTION
**Pull request by I-Bouros**
Created abstract ForwardModel Class as in issue #7 with appropriate Unittest.

### CODE REVIEW

File: branchpro/tests/test_models.py [hunk 1]
```diff
@@ -6,9 +6,13 @@ class TestForwardModelClass(unittest.TestCase):
...
-    def test_output(self):
-        forwardModel = bp.ForwardModel()
-        self.assertEqual(forwardModel.value, 1)
+    def test__init__(self):
+        bp.ForwardModel()
+
+    def test_simulate(self):
+        forward_model = bp.ForwardModel()
+        with self.assertRaises(NotImplementedError):
+            forward_model.simulate()
```
Comment (rcw5890) [inline branchpro/tests/test_models.py:15]: Simulate() needs parameters and times

**Review by rcw5890 — COMMENTED** [review]
Looks great, can you check on TestForwardModelClass.test_simulate ?
Thanks

**Review by DavAug — APPROVED** [review]
This looks really good @I-Bouros!
Thank you for making the changes.

**Review by rcw5890 — APPROVED** [review]
I merged the new tests that @ao20 and I were working on, and it all passes. 

Feel free to go ahead and merge and delete this branch

**Pull request merged**